go-controller/pkg/util/util.go

```diff
@@ -393,3 +393,20 @@ func NoHostSubnet(node *v1.Node) bool {
 	nodeSelector, _ := metav1.LabelSelectorAsSelector(config.Kubernetes.NoHostSubnetNodes)
 	return nodeSelector.Matches(labels.Set(node.Labels))
 }
+
+// GetNBZone returns the zone name configured in the OVN Northbound database.
+// If the zone name is not configured, it returns the default zone name - "global"
+// It retuns error if there is no NBGlobal row.
+func GetNBZone(nbClient libovsdbclient.Client) (string, error) {
```

Comment:
This function defaults to default global zone if the zone name is not set.

But it will return error if there is no NB_Global row in the DB (which is likely to happen only if ovn-northd fail to create one) or if some connection issues to NB DB.

You mean its ok to return default global zone for these errors too ?

Reply:
The zone name in NB DB will be set by a prestart hook right so it should always be there before NCM starts?

>You mean its ok to return default global zone for these errors too ?

No, we must return err for those scenarios and that is what is happening in your code. I am just wondering about the scenario of NB Global table column name has an empty entry of "". So even if its empty and we are using multi zone, we wont continue having a global zone because we compare with the zone cli arg.

Nothing to do here numan.. just trying to understand this.

Reply:
> The zone name in NB DB will be set by a prestart hook right so it should always be there before NCM starts?
> 

Yes.  That's my assumption.  It should be set by prestart hook.  But it could be set in the nbdb container and may take a bit of time to get propagated if NCM starts before the nbdb 


> > You mean its ok to return default global zone for these errors too ?
> 
> No, we must return err for those scenarios and that is what is happening in your code. I am just wondering about the scenario of NB Global table column name has an empty entry of "". So even if its empty and we are using multi zone, we wont continue having a global zone because we compare with the zone cli arg.
> 
> Nothing to do here numan.. just trying to understand this.